### PR TITLE
Fix slicing handling

### LINKED ISSUE #269: refactor vdx slices
simplify the slice lookup routines

### PR DESCRIPTION
**Pull request by nbassler**
Closes #269.

Needed by https://github.com/pytrip/pytripgui/pull/75

### CODE REVIEW

File: pytrip/vdx.py [hunk 1]
```diff
@@ -867,19 +867,15 @@ def read_dicom(self, info, data):
...
-            return 3  # TODO: what is this? And shoudn't it be float?
-        return abs(float(self.slice_z[1]) - float(self.slice_z[0])) / 100
+            # get current slice position
+            _z_pos = contour.ContourData[2]
+
+            # if we have a new z_position, add a new slice object to self.slices
+            if _z_pos not in [i.get_position() for i in self.slices]:
+                logger.debug("Append new slice at z_position {:f} to slices list:".format(_z_pos))
+                self.slices.append(Slice(cube=self.cube))
+                self.slices[-1].add_dicom_contour(contour)
```
Comment (grzanka) [inline pytrip/vdx.py]: This looks a bit enigmatic, what about following code ?
```
s = Slice(cube=self.cube)
s.add_dicom_contour(contour)
self.slices.append(s)
```

I think it is a bit easier to debug.

**Comment by grzanka** [comment]
I like it, its very pythonic !

File: pytrip/vdx.py [hunk 2]
```diff
@@ -952,8 +949,8 @@ def concat_contour(self):
         """ Concat all contours in all slices found in this VOI.
         """
         if not self.is_concated:
-            for k in self.slices.keys():
-                self.slices[k].concat_contour()
+            for sl in self.slices:
+                sl.concat_contour()  # TODO: check scope
```
Comment (nbassler) [inline pytrip/vdx.py]: @grzanka hope changes to sl. are not local to the function, right?

**Comment by grzanka** [comment]
tests are failing, see `tests/test_vdx.py` file, it fails on:
```
self.assertEqual(target_voi.get_thickness(), 3)
```

File: pytrip/vdx.py (hunk 2 above)
Comment (grzanka) [inline pytrip/vdx.py]: it should

File: pytrip/vdx.py (hunk 1 above)
Comment (nbassler) [inline pytrip/vdx.py]: agree.

**Comment by grzanka** [comment]
Tests still failing, see https://travis-ci.org/pytrip/pytrip/jobs/205923656

```
self = <pytrip.vdx.Voi instance at 0x7fc5fbd03c20>
    def concat_to_3d_polygon(self):
        """ Concats all contours into a single contour, and writes all data points to sefl.polygon3d.
            """
        self.concat_contour()
        data = []
        for slice in self.slices:
>           data.extend(self.slices[slice].contour[0].contour)
E           TypeError: object cannot be interpreted as an index
```

**Comment by grzanka** [comment]
Still failing https://travis-ci.org/pytrip/pytrip/jobs/205952607

```
test_create_voi_sphere
(...)
        logger.info("Checking Voi get_slice_at_pos method, good slice")
        s = v.get_slice_at_pos(11)
>       self.assertIsNotNone(s)
E       AssertionError: unexpectedly None
```

**Pull request merged**